### ISSUE TITLE
The 'cmake-server(7)' is deprecated.

### ISSUE THREAD

**Issue by Colengms** [microsoft/vscode-cmake-tools#710]
Perhaps obvious, but just wanted to point out that with CMake 3.15 installed, I get the following warning when configuring:

CMake Deprecation Warning:
  The 'cmake-server(7)' is deprecated.  Please port clients to use the
  'cmake-file-api(7)' instead.

**Comment by jnickborys** [comment]
Bumping this. I hit it too.

**Comment by rustedwizard** [comment]
Same here.

**Comment by Ian-Re** [comment]
same here.

**Comment by JesvinJoseph** [comment]
same here

**Comment by tkit1994** [comment]
Same here.

**Comment by cestlascorpion** [comment]
same here. And how can I use the  "cmake-file-api(7)" instead?

**Comment by Zingam** [comment]
You can't. That's a task for the extension to implement. I have looked up the sources and I am not sure if that would be a trivial task. So I don't expect that to change soon but for now I don't think that this warning message should bother you.

**Comment by RTS007** [comment]
Same here.

**Comment by Toxe** [comment]
In the meantime, is there any way to disable this warning?

**Comment by Randshot** [comment]
From what I know, this requires significant changes in core parts of the extension, so I think that we won't see any update on this soon, depending on how many people are actively involved in maintaining this extension.

**Comment by bobbrow** [comment]
I believe @KoeMai's PR implements this https://github.com/microsoft/vscode-cmake-tools/pull/720.  I haven't had a good chunk of time to review it yet, but it's on my short list.

**Comment by Randshot** [comment]
Oh, didn't see that one. Looks promising, but like I expected a rather large effort.

**Comment by nmoinvaz** [comment]
Uninstalling cmake 3.15 and installing the previous version 3.14.6 appears to work for me. Obviously its not a long term solution.

**Comment by eddyekofo94** [comment]
Any fix thus far?

**Comment by eddyekofo94** [comment]
Make has [this](https://cmake.org/cmake/help/v3.15/release/3.15.html#deprecated-and-removed-features) one their CMake 3.15 Release Notes.

**Comment by roy-tian** [comment]
I put this line in settings.json and the warning disppears:
`"cmake.useCMakeServer": false`

**Comment by Zingam** [comment]
Off-topic: Is anybody running VSCode + CMake 3.15 (particularly CMake form the snap package) on Ubuntu?

**Comment by timxian** [comment]
> Off-topic: Is anybody running VSCode + CMake 3.15 (particularly CMake form the snap package) on Ubuntu?

here

**Comment by FISHFISHFISHLOVECAT** [comment]
Same here

**Comment by HeyTech** [comment]
Are there a fix for this?

**Comment by roy-tian** [comment]
I set "cmake.useCMakeServer" to "false" and the warning was gone.

**Comment by bobbrow** [comment]
The warning can be ignored for now.  It doesn't harm the experience.  We plan to add support for cmake-file-api soon.

**Comment by Zingam** [comment]
> I set "cmake.useCMakeServer" to "false" and the warning was gone.

@roy-tian Please, don't do that! You've just ruined your whole experience :)

**Comment by siriobalmelli** [comment]
The problem at this time seems to be:

## Setting `"cmake.useCMakeServer": false`:

- I get no list of build targets
- Manually entering a build target builds cleanly
- I cannot seem to find an option to *manually* specify targets in `settings.json`; I have looked [here](https://vector-of-bool.github.io/docs/vscode-cmake-tools/building.html) and [here](https://vector-of-bool.github.io/docs/vscode-cmake-tools/settings.html)

## Setting `"cmake.useCMakeServer": true`:

- I *do* get a list of build targets
- I get a crash when trying to run a build:
```
[cms-client] The connection to cmake-server was terminated unexpectedly
[cms-client] cmake-server exited with status null (SIGSEGV)
```

## Notes

It appears that the cmake server may not behave the same as the file API?
See [here](http://cmake.3232098.n2.nabble.com/The-connection-to-cmake-server-was-terminated-unexpectedly-cms-client-cmake-server-exited-with-statu-td7599949.html) for an example.

Unfortunately I cannot edit the `CMakeLists.txt` in question because:

- it is a [manufacturer's SDK](https://www.nordicsemi.com/Software-and-Tools/Software/nRF5-SDK-for-Mesh)
- it does legitimately work when running `cmake` on the command line.

This may possibly be related to:

- https://github.com/microsoft/vscode-cmake-tools/issues/752
- https://github.com/microsoft/vscode-cmake-tools/issues/688

## Conclusion

Perhaps the simplest solution for now is to provide a configuration setting to manually specify targets in `settings.json`?

I'm at an impasse with this, any recommendations are welcome.

**Comment by Zingam** [comment]
@siriobalmelli If you are experience a crash, you should create a new issue. This feels like a CMake bug to me, so maybe report it there too.

File API is an alternative (new) way of IDEs to communicate with CMake. This warning is **not a real issue**, so you should **just ignore it**. I don't know if it can be disabled. The next version will probably support File API so it is not worth bothering fixing this warning.

**Comment by grownanny** [comment]
same issue when setting up make is VScode

**Comment by bobbrow** [comment]
This was addressed in 1.3.0
